sale_commission_check_deposit/wizard/wizard_check_settle.py

```diff
@@ -0,0 +1,47 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+import json
+from datetime import date, timedelta
+
+from odoo import models
+
+
+class SaleCommissionMakeSettle(models.TransientModel):
+    _inherit = "sale.commission.make.settle"
+
+    def _get_agent_lines(self, agent, date_to_agent):
+        """
+        Exclude invoices with checks payment if at least safety days
+        haven't passed since expiration date set on check deposit.
+        """
+        agent_lines = super()._get_agent_lines(agent, date_to_agent)
+        for line in agent_lines:
+            if (
+                line.commission_id.invoice_state == "paid"
+                and line.invoice_id.invoice_payments_widget != "false"
+            ):
+                move_ids = self.get_move_ids(line.invoice_id.invoice_payments_widget)
+                for move in move_ids:
+                    check_deposit_line = move.line_ids.filtered(
+                        lambda x: x.check_deposit_id
+                    )
+                    if not check_deposit_line or (
+                        check_deposit_line.date_maturity
+                        + timedelta(days=check_deposit_line.journal_id.safety_days)
+                        > date.today()
+                    ):
+                        agent_lines -= line
+                        break
+        return agent_lines
+
+    def get_move_ids(self, payments_widget):
+        move_dict = json.loads(payments_widget)
+        account_move_ids = [content["move_id"] for content in move_dict["content"]]
+        move_lines = self.env["account.move.line"].search(
+            [
+                ("move_id", "in", account_move_ids),
+                ("journal_id.is_check_journal", "=", True),
+            ]
+        )
+        return move_lines.mapped("move_id")
```

Comment:
If function returns only move_lines with check_deposit_id it will not remove agent line without it.

Reply:
if needed, you can always ref by `move_line.move_id`, isn't it?

Reply:
or return `move_lines.filtered(lambda x: x.check_deposit_id).mapped("move_id")`

Reply:
or even: 
```
move_lines.filtered(lambda x: x.check_deposit_id and x.check_deposit_line.date_maturity
                        + timedelta(days=check_deposit_line.journal_id.safety_days)
                        < date.today()).mapped("move_id")
```

so that u return only move having required conditions

Reply:
Yes, but I also need move_id without check_deposit_id because I need to remove that agent_line. If I filter to get only move_lines with check_deposit_id how I can't remove that agent_lines.

Reply:
sorry not getting the point .. u have all lines and all lines that match conditions, what about removing alla at once like all_lines - all lines that match conditions = left lines to remove. Am I missing something?  

Other than that u are removing 1 line and than breaking loop .. 

surely i'm missing something, lets wait some other review

Reply:
Sorry I understand only now what you mean. I will correct now. Thanks